src/resmom/catch_child.c

```diff
@@ -1915,164 +1913,54 @@ scan_for_exiting(void)
 }
 
 /**
- * @brief
- * 	send old style IS_RESTART message to Server.
- *	Used when Server is older & does not recognize the TCP Restart message.
+ * @brief	Send a restart message to the Server over IS protocol
  *
  * @par
- *	Open an RPP stream to the named server/port, compose the IS_RESTART,
- *	flush the stream and then close it.
- *
- * @param[in]	svr  - name of Server to which to send the restart
- * @param[in]	port - port Server would be expecting to receive IM messages
- *
- * @return	void
+ *	Close any existing tpp streams to the server, it is unlikely that
+ *	there is one. Parse the server name from pbs.conf;
+ *	Use PBS_SERVER_HOST_NAME if defined, else use PBS_SERVER,
+ *	open TPP stream and send restart on it
  *
+ * @return void
  */
-
-static void
-send_restart_rpp(char *svr, unsigned int port)
+void
+send_restart(void)
 {
-	int		j;
+	unsigned int port = default_server_port;
+	char *svr;
+	int j;
+
+	if (server_stream >= 0) {
+		sprintf(log_buffer, "Closing existing server stream %d", server_stream);
+		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_NOTICE, msg_daemonname, log_buffer);
```

Comment:
consider using log_eventf()

Reply:
Not now, but in the future, we should define functions that takes variable arguments like log_eventf for all other log functions like log_err, log_record, etc...

Reply:
Yeah good idea.  I implemented this first for log_event() because it was the most used function.  I converted all of the scheduler's calls into log_eventf().  I'm not sure we need log_record() because the only real difference is that log_event() looks at the log_events attribute, and log_record() has no such filter (log_event() actually calls log_record()).  Doing a log_event() with force is really the same thing.  Having a log_errf() would be nice.